src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1262,6 +1263,28 @@ public static Process Start(string fileName, string arguments)
             return Start(new ProcessStartInfo(fileName, arguments))!;
         }
 
+        /// <devdoc>
+        ///    <para>
+        ///       Starts a process resource by specifying the name of an
+        ///       application and a set of command line arguments. Associates the process resource
+        ///       with a new <see cref='System.Diagnostics.Process'/>
+        ///       component.
+        ///    </para>
+        /// </devdoc>
+        public static Process Start(string fileName, IEnumerable<string> arguments)
+        {
+            if (arguments == null)
```

Comment:
Why do we check arguments but not filename?

Reply:
because other overloads of `Start` and `ProcessStartInfo` ctor allow for `fileName == null`:

https://github.com/dotnet/runtime/blob/ca5062c451e873750f9b221f32e1f42f8fd53694/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L1258-L1263

https://github.com/dotnet/runtime/blob/ca5062c451e873750f9b221f32e1f42f8fd53694/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs#L42-L45

Reply:
But those also allow arguments to be null.

Reply:
Also, presumably this `IEnumerable<string> arguments` and the `string arguments` in the previous overload should be nullable? (I'm not sure why we don't validate `fileName` as being non-null, but presumably Start will actually fail if it is null.)
cc: @eiriktsarpalis

Reply:
> But those also allow arguments to be null

You are right. Since this is a new API we should most probably not allow for nulls at all?

Reply:
I added explicit null checks for both arguments. There is no reason for allowing nulls in this overload